pkg/cmd/cli/cmd/importimage.go

```diff
@@ -98,6 +99,15 @@ func RunImportImage(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, arg
 
 	if stream.Annotations != nil {
 		delete(stream.Annotations, imageapi.DockerImageRepositoryCheckAnnotation)
+		if insecure {
+			stream.Annotations[imageapi.InsecureRepositoryAnnotation] = "true"
+		}
+	} else {
+		if insecure {
+			annotations := make(map[string]string)
+			stream.Annotations = annotations
+			stream.Annotations[imageapi.InsecureRepositoryAnnotation] = "true"
+		}
```

Comment:
Instead of repeating the insecure if check do it after the if-else setting annotations, and it'll be only once..